packages/preview-middleware-client/src/adp/controllers/ControllerExtension.controller.ts

```diff
@@ -0,0 +1,203 @@
+/** sap.m */
+import Input from 'sap/m/Input';
+import Button from 'sap/m/Button';
+import type Dialog from 'sap/m/Dialog';
+import MessageToast from 'sap/m/MessageToast';
+
+/** sap.ui.core */
+import { ValueState } from 'sap/ui/core/library';
+import type UI5Element from 'sap/ui/core/Element';
+import Controller from 'sap/ui/core/mvc/Controller';
+
+/** sap.ui.base */
+import type Event from 'sap/ui/base/Event';
+
+/** sap.ui.model */
+import JSONModel from 'sap/ui/model/json/JSONModel';
+
+/** sap.ui.rta */
+import type RuntimeAuthoring from 'sap/ui/rta/RuntimeAuthoring';
+
+/** sap.ui.fl */
+import Utils from 'sap/ui/fl/Utils';
+
+/** sap.ui.dt */
+import type ElementOverlay from 'sap/ui/dt/ElementOverlay';
+
+import type { ControllersResponse } from '../api-handler';
+import { readControllers, writeChange, writeController } from '../api-handler';
+
+interface ControllerExtensionService {
+    add: (codeRef: string, viewId: string) => Promise<unknown>;
+}
+
+/**
+ * @namespace open.ux.preview.client.adp.controllers
+ */
+export default class ControllerExtension extends Controller {
```

Comment:
Very similar to the AddFragment.controller, therefore, please consider a base controller consolidating common properties and functions

Reply:
Changed in: https://github.com/SAP/open-ux-tools/pull/1275/commits/35d8a13cea5cf636185d1d7ff2a5c85e46e54080